File: Resueltos_por_competidor/Miguel_Lopez/Stickogon.cpp
/*
 * Autor: Miguel Lopez
 * Problema: Stickogon
 * Juez online: CodeForces
 * Veredicto: Accepted
 * Url: https://codeforces.com/problemset/problem/1957/A
 */

// Source: https://usaco.guide/general/io

#include <bits/stdc++.h>
using namespace std;

int main() {
	int t; cin >> t;
    while(t--) {
        int i; cin >> i;
        map<int,int> ca;
        int cont = 0;
        while(i--) {
            int z; cin >> z;
            ca[z]++;
        }
        for(auto[val, cant]: ca) {
            cont+=cant/3;
        }
        cout << cont << endl;
        ca.clear();
        }
    }

File: Resueltos_por_competidor/Gabriella_Castro/Stickogon.java
/*
 * Autor: Gabriella Castro
 * Problema: Stickogon
 * Juez online: VJudge
 * Veredicto: Accepted
 * URL: https://vjudge.net/problem/CodeForces-1957A
 */

import java.util.HashMap;
import java.util.Scanner;

public class Main {

	public static void main(String[] args) {

		Scanner sc = new Scanner(System.in);

		int t = sc.nextInt();
		HashMap<Integer, Integer> frecuencias = new HashMap<>();

		for (int i = 0; i < t; i++) {
			int conta = 0;
			int n = sc.nextInt();
			frecuencias = new HashMap<>();
			for (int j = 0; j < n; j++) {
				int num = sc.nextInt();
				frecuencias.put(num, frecuencias.getOrDefault(num, 0) + 1);
			}

			for (int entrada : frecuencias.values()) {
				if (entrada > 2) {
					conta +=(int) Math.floor((double)entrada/3);

				}

			}
			System.out.println(conta);

		}

	}

}
